includes/controllers/class-user.php

```diff
@@ -522,7 +537,7 @@ public function request_user_password( $request ) {
 
 		// Check email key.
 		if ( get_option( 'hp_user_verify_email' ) && $user_object->hp_email_verify_key ) {
-			return hp\rest_error( 401, esc_html__( 'Please check your email to activate your account.', 'hivepress' ) );
+			return hp\rest_redirect( hivepress()->router->get_url( 'user_resend_email_verify_page', [ 'user_id' => $user->get_id() ] ) );
```

Comment:
It's ok to show the error text in the login form, I meant adding redirect to the registration form (to a new route you created), most users will be redirected there after registration and if they don't get an email they'll already have this form for re-sending available.